server/src/main/java/org/jboss/as/server/deployment/RootDeploymentUnitService.java

```diff
@@ -51,7 +53,7 @@ public RootDeploymentUnitService(final Consumer<DeploymentUnit> deploymentUnitCo
                                      final Supplier<DeploymentMountProvider> serverDeploymentRepositorySupplier,
                                      final Supplier<PathManager> pathManagerSupplier,
                                      final Supplier<VirtualFile> contentsSupplier,
-                                     final String name, final String managementName, final DeploymentUnit parent,
+                                     final String name, final String managementName, final DeploymentUnit parent, Stability stability,
```

Comment:
I would suggest adding `Final` or removing all the `Final` modifiers.

In general, it is better if we follow the same pattern, that will avoid any doubts or questions about why ones are final and not others.

Reply:
Updated